storage/include/sparse_merkle/internal_node.h

```diff
@@ -202,7 +203,10 @@ class BatchedInternalNode {
 
   // A remove has failed, because the the leaf is further down the tree.
   // Instruct the caller to try the next node.
-  struct Descend {};
+  struct Descend {
+    // The version of the next BatchedInternalNode further down the tree.
+    Version version;
```

Comment:
[Minor]: Maybe we can name the version so that it is a bit clearer, e.g. `next_batched_node_version` or similar.